crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -156,6 +156,22 @@ def compute_resources_available?
         true
       end
 
+      # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+      def ha_presence_check
```

Comment:
Splitting functions into smaller parts is almost always a good idea! And sounds like it makes sense to me In this case.